src/SCRIPTS/BF/ui.lua

```diff
@@ -266,6 +267,19 @@ end
 
 function run_ui(event)
     local now = getTime()
+    
+    if not dataInitialised then
+        if data_init == nil then
+          data_init = assert(loadScript(SCRIPT_HOME .. "/data_init.lua"))()
+        end
+        dataInitialised = data_init.init();
```

Comment:
Why are you running this only once? If you want to run this here, it should be run until the initialisation has finished, otherwise this will just be done after an arbitrary number of user interactions.